python/GafferArnoldUI/ArnoldShaderUI.py

```diff
@@ -339,7 +339,7 @@ def __plugMetadata( plug, name ) :
 		key = plug.node()["name"].getValue() + "." + plug.relativeName( node )
 	else :
 		# Node type is ArnoldLight.
-		key = plug.node()["__shaderName"].getValue() + "." + plug.relativeName( node )
+		key = plug.node()["__shader"]["name"].getValue() + "." + plug.relativeName( node )
```

Comment:
Now we have the `__parameterMetadata()` forwarding implemented in ArnoldLightUI, why do we need a branch for lights here at all?

Reply:
Actually, I'm pretty sure we _do_ want this, and it's the forwarding in ArnoldLightUI that is superfluous. See comments on that file.